tests/integration/pilot/common/routing.go

```diff
@@ -445,23 +491,16 @@ func protocolSniffingCases(apps *EchoDeployments) []TrafficTestCase {
 						// TODO(https://github.com/istio/istio/issues/26798) enable sniffing tcp
 						skip: call.scheme == scheme.TCP,
 						name: fmt.Sprintf("%v %v->%v from %s", call.port, client.Config().Service, destination.Config().Service, client.Config().Cluster.Name()),
-						call: func() (echoclient.ParsedResponses, error) {
-							return client.Call(echo.CallOptions{
-								Target:   destination,
-								PortName: call.port,
-								Scheme:   call.scheme,
-								Count:    callCount,
-								Timeout:  time.Second * 5,
-							})
-						},
-						validator: func(responses echoclient.ParsedResponses) error {
-							if err := responses.CheckOK(); err != nil {
-								return err
-							}
-							if err := responses.CheckHost(destination.Config().HostHeader()); err != nil {
-								return err
-							}
-							return nil
+						call: client.CallWithRetryOrFail,
+						opts: echo.CallOptions{
+							Target:   destination,
+							PortName: call.port,
+							Scheme:   call.scheme,
+							Count:    callCount,
+							Timeout:  time.Second * 5,
+							Validator: echo.And(
```

Comment:
why use echo.And vs echo.ExpectOK().And(..)?

Reply:
Since everyone copy+pastes tests, want to make sure we are doing the "best practices"

Reply:
A couple reasons ...

1. Having `And` as a method on `Validator` requires that each `Validator` implements it.
2. `Validator.And` makes chaining harder to read. Using `And` as a global function makes the list of validators more clear.